airflow/providers/amazon/aws/operators/redshift_cluster.py

```diff
@@ -676,10 +685,26 @@ def execute(self, context: Context):
                     time.sleep(self._attempt_interval)
                 else:
                     raise
-
-        if self.wait_for_completion:
+        if self.deferrable:
+            self.defer(
+                timeout=self.execution_timeout,
```

Comment:
There was a [discussion](https://github.com/apache/airflow/pull/30853#discussion_r1187018026) about this `timeout` parameter, and the solution we ended up going with was to use the `self.max_attempts` and `self.poll_interval` along with an extra 60 seconds to set the timeout from the operator side.